src/legendoptics/vm2000.py

```diff
@@ -0,0 +1,260 @@
+            vm2000_reflectivity[ji] = 0.95  # Visible light 0.95, 0.99
+        else:
+            vm2000_reflectivity[ji] = 0.12  # UV light 0.15, 0.3 (paper)
+
+        if vm2000_energy_range[ji] > 3.35 * u.eV:  # 5 eV 3.35
+            # depending on path length in foil --> angle
+            wls_absorption[ji] = vm2000_calculate_wls_mfp(wls_yield)  # Absorbs UV
+        else:
+            wls_absorption[ji] = (
+                1.0 * u.m
+            )  # Imperturbed, no absorption of visible light
+
+    g = InterpolatingGraph(
+        *readdatafile("vm2000_em_spec.dat", pkg="l200geom.materials"), zero_outside=True
+    )
+    wls_emission = g(vm2000_energy_range.to("nm")).to("dimensionless")
+
+    # Copy the first element to 0th position
+    wls_absorption[0] = wls_absorption[1]  # depending on path length in foil --> angle
+    wls_emission[0] = wls_emission[1]
+
+    return (
+        vm2000_energy_range,
+        vm2000_reflectivity,
+        vm2000_efficiency,
+        wls_absorption,
+        wls_emission,
+    )
+
+
+@store.register_pluggable
+def vm2000_scint_timeconstant() -> Quantity:
+    """Time constant, from MaGe.
+
+    .. optics-const::
+    """
+    return 0.5 * u.ns
+
+
+def pyg4_vm2000_attach_rindex(mat, reg) -> None:
+    """Attach the refractive index to the given VM2000 material instance.
+
+    See Also
+    --------
+    .vm2000_refractive_index
+    """
+    λ = np.array([100, 600]) * u.nm
+    r = [vm2000_refractive_index()] * 2
+
+    with u.context("sp"):
+        mat.addVecPropertyPint("RINDEX", λ.to("eV"), r)
+
+
+def pyg4_vm2000_attach_absorption_length(mat, reg) -> None:
+    """Attach the refractive index to the given VM2000 material instance.
+
+    See Also
+    --------
+    .vm2000_absorption_length
+    """
+    λ = np.array([100, 600]) * u.nm
+    r = [vm2000_absorption_length()] * 2 * u.m
+
+    with u.context("sp"):
+        mat.addVecPropertyPint("ABSLENGTH", λ.to("eV"), r)
+
+
+def pyg4_vm2000_attach_particle_scintillationyields(mat, reg) -> None:
+    """Attach the scintillation yiels (except of electron yield) to the given VM2000 material instance.
+
+    See Also
+    --------
+    .vm2000_particle_scintillationyields
+    """
+    energy = vm2000_particle_scintillationyields()[0] * u.eV
+    proton_yield = vm2000_particle_scintillationyields()[1]
+    alpha_yield = vm2000_particle_scintillationyields()[2]
+    deuteron_yield = vm2000_particle_scintillationyields()[3]
+    ion_yield = vm2000_particle_scintillationyields()[4]
+    triton_yield = vm2000_particle_scintillationyields()[5]
+
+    mat.addVecPropertyPint("PROTONSCINTILLATIONYIELD", energy, proton_yield)
+    mat.addVecPropertyPint("ALPHASCINTILLATIONYIELD", energy, alpha_yield)
+    mat.addVecPropertyPint("DEUTERONSCINTILLATIONYIELD", energy, deuteron_yield)
+    mat.addVecPropertyPint("IONSCINTILLATIONYIELD", energy, ion_yield)
+    mat.addVecPropertyPint("TRITONSCINTILLATIONYIELD", energy, triton_yield)
```

Comment:
we have a helper function to set the particle scintillation yield. In this cases, I would also avoid the seperate function defintion above, as those values are all just zeros.

```suggestion
    pyg4utils._def_scint_particle(mat, "PROTON", 0 / u.eV, None)
    pyg4utils._def_scint_particle(mat, "ALPHA", 0 / u.eV, None)
    pyg4utils._def_scint_particle(mat, "DEUTERON", 0 / u.eV, None)
    pyg4utils._def_scint_particle(mat, "ION", 0 / u.eV, None)
    pyg4utils._def_scint_particle(mat, "TRITON", 0 / u.eV, None)
```